opacus/grad_sample/functorch.py

```diff
@@ -1,5 +1,40 @@
+import copy
+
+import torch
 import torch.nn as nn
 from opacus.layers.dp_rnn import RNNLinear
+from torch.func import grad, vmap
+
+
+# https://gist.github.com/zou3519/7769506acc899d83ef1464e28f22e6cf
+def make_functional(mod: nn.Module, disable_autograd_tracking: bool = False):
```

Comment:
Should we try to integrate this a bit more to Opacus? Maybe in a later PR?

Reply:
You mean using the native `torch.func.functional_call` in Opacus?

I'm not sure it'll look significantly different from the current solution. The key part of `make_functional` is deep copying the model, which we'll need to do anyway. The rest is very similar to how *native* API would look like